hs/WebParsing/TimeTableParser.hs

```diff
@@ -36,38 +39,32 @@ insertDepartment :: MonadIO m => ([T.Text], T.Text) -> ReaderT SqlBackend m ()
 insertDepartment (code, name) = 
     insert_ (Tables.Department code name)
 
--- | List of unwanted words to remove when getting department codes.
-wordsToRemove :: [String]
-wordsToRemove = ["courses",
-                "courses]", 
-                "includes",
-                "and",
-                "-",
-                "[",
-                "]",
-                ""]
+-- | Extracts a list of lists of department names followed by unfiltered department codes.
+splitTagsToLists :: String -> Either ParseError [[String]]
+splitTagsToLists input =
+    let tags = endBy line eol
+        line = sepBy cell (char '[')
```

Comment:
Thanks! I've changed extractDeptInfo to match the type signature you mentioned and simplified the code. I'll look into the manyTill and oneOf functions. It's an interesting solution!